app.js

```diff
@@ -50,9 +52,52 @@ export default class App extends Component {
     })
   }
 
+  handleEditCancel() {
+    this.setState({isEditing: false})
+  }
+
+  handleEdit(event) {
+    event.preventDefault()
+    const {data} = this.state
+    const id = parseInt(event.target.closest('div').id)
+    this.setState({
+      isEditing: true,
```

Comment:
Be ware of `Boolean` flags, you can usually make decisions based off of other information.